### DISCUSSION TITLE
color blocks appear on the screen

### DISCUSSION THREAD

**Discussion by brawoh** [rustdesk/rustdesk#8916] [Q&A] [answered]
I didn't have any problems with the 1.2.6 version, but when I upgraded to the 1.2.7 version today, the screen showed pink and green blocks. I didn't change the preferences.
![2024-08-01_155540](https://github.com/user-attachments/assets/2925157f-f227-4eeb-b1a3-d27ab31b1d43)

**Comment by 21pages** [comment]
Is it always like this,  or it will disappear after a while? Which codec?

**Reply by brawoh** [reply]
When switching windows, there will be fewer color blocks, and if  keep  window for a long time, there will be more.Codec I set automatically

**Reply by 21pages** [reply, marked as answer]
Please try different codecs, you can see the current codec with "Show quality monitor" selected.

![1722503115069](https://github.com/user-attachments/assets/6f74689d-75fb-4494-83e8-aa713bf36004)

**Reply by brawoh** [reply]
Very helpful, thank you, VP9 is automatically selected. When I switch to VP8 or AV1, the color block disappears.

**Reply by 21pages** [reply]
> I didn't have any problems with the 1.2.6 version, but when I upgraded to the 1.2.7 version today

Maybe libvpx needs to be downgraded, 1.14.1 -> 1.13.1

**Reply by 21pages** [reply]
@brawoh Could you try https://github.com/21pages/rustdesk/releases/download/nightly/rustdesk-1.3.0-x86_64-libvpx-1.31.1.exe? This program downgrade libvpx.

**Reply by brawoh** [reply]
Ok, I have tried this program and it works, even VP9 has no color blocks anymore.

**Reply by 21pages** [reply]
Will downgrade if more feedback is received.

**Reply by basilgello** [reply]
Please note that downgrading libvpx without applying additional patches will reintroduce CVE that I aimed to fix. We need to check which patches to add and if they break things first. If they do, better to file a bug against upstream.

**Reply by basilgello** [reply]
These patches need to be backported to 1.13  (according to [Debian security tracker](https://security-tracker.debian.org/tracker/CVE-2024-5197)):

https://github.com/webmproject/libvpx/commit/c5640e3300690705c336966e2a8bb346a388c829
https://github.com/webmproject/libvpx/commit/9d7054c0cb83665a74cf6f59b6261f455e692149
https://github.com/webmproject/libvpx/commit/61c4d556bd03b97d84e3fa49180d14bde5a62ba
